src/api/auth.ts

```diff
@@ -134,3 +134,234 @@ function anonymousUser(): any {
 function authenticatedUser(name: string, groups: string[] = []): any {
   return { name: name, groups: groups.concat(['&all', '$anonymous']), real_groups: groups };
 }
+
+export function newUser(req: express.Request, res: express.Response): express.Response {
+  let username = req.body.username;
+  let fullname = req.body.fullname;
+  let email = req.body.email;
+  let password = req.body.password;
+  let auth = getAuth();
+  let config = getConfig();
+  let hash = generateHash(password, config.secret);
+  let user = {
+    'name': username,
+    'password': hash,
+    'fullName': fullname,
+    'email': email
+  };
+  let index = auth.users.findIndex(u => u.name === username);
+  if (index === -1) {
+    auth.users.push(user);
+    writeJsonFile(getAuthPath(), auth);
```

Comment:
isn't this a promise base function, this would never get written..
```ts
writeJsonFile(getAuthPath, auth).then(() => res.status(200).json({ success: 'User successfully added.' });
```
..would be okay.